bin/tx-prover/README.md

```diff
@@ -1,48 +1,60 @@
+meant to handle multiple workers with a queue for each one.
 
 ## Installation
 
-Install the prover binary for production using `cargo`:
+Install the prover worker and proxy binaries for production using `cargo`:
 
 ```sh
-cargo install miden-tx-prover --locked
+cargo install miden-tx-prover-worker --locked
+cargo install miden-tx-prover-proxy --locked
 ```
 
 This will install the latest official version of the prover. You can install a specific version using `--version <x.y.z>`:
 
 ```sh
-cargo install miden-tx-prover --locked --version x.y.z
+cargo install miden-tx-prover-worker --locked --version x.y.z
+cargo install miden-tx-prover-proxy --locked --version x.y.z
 ```
 
-You can also use `cargo` to compile the prover service from the source code if for some reason you need a specific git revision. Note that since these aren't official releases we cannot provide much support for any issues you run into, so consider this for advanced users only. The incantation is a little different as you'll be targetting this repo instead:
+To install both services at once from the source code using specific git revisions you can use `cargo install`. Note that since these aren't official releases we cannot provide much support for any issues you run into, so consider this for advanced users only:
 
 ```sh
 # Install from a specific branch
-cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover --branch <branch>
+cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover-worker --branch <branch>
+cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover-proxy --branch <branch>
 
 # Install a specific tag
-cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover --tag <tag>
+cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover-worker --tag <tag>
+cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover-proxy --tag <tag>
 
 # Install a specific git revision
-cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover --rev <git-sha>
+cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover-worker --rev <git-sha>
+cargo install --locked --git https://github.com/0xPolygonMiden/miden-base miden-tx-prover-proxy --rev <git-sha>
 ```
-
-If you want to build the prover from a local version, from the root of the workspace you can run:
+If you want to build from a local version, from the root of the workspace you can run:
 
 ```bash
-make install-prover
+make install-prover-worker
+make install-prover-proxy
 ```
 
-This step will also generate the necessary protobuf-related files.
+Note that for the prover worker you might need to add the `testing` feature in case you need. This step will also generate the necessary protobuf-related files.
```

Comment:
```suggestion
Note that for the prover worker you might need to enable the `testing` feature in case the transactions were executed with reduced proof-of-work requirements (or otherwise, the proving process will fail). This step will also generate the necessary protobuf-related files.
```

Reply:
Also, maybe we should mention that it can be installed with `make install-prover-worker-testing` while we are explaining this